src/ukify/ukify.py

```diff
@@ -572,12 +572,22 @@ def pe_add_sections(uki: UKI, output: str):
         else:
             new_section.IMAGE_SCN_CNT_INITIALIZED_DATA = True
 
-        pe.__data__ = pe.__data__[:] + bytes(new_section.PointerToRawData - len(pe.__data__)) + data + bytes(new_section.SizeOfRawData - len(data))
+        # Special case, mostly for .sbat: the stub will already have a .sbat section, but we want to append
+        # the one from the kernel to it. It should be small enough to fit in the existing section, so just
+        # swap the data.
+        for i, s in enumerate(pe.sections):
+            if s.Name.rstrip(b"\x00").decode() == section.name and new_section.Misc_VirtualSize <= s.SizeOfRawData:
+                pe.__data__ = pe.__data__[:s.PointerToRawData] + data +  pe.__data__[pe.sections[i+1].PointerToRawData:]
```

Comment:
This doesn't append the sbat, it replaces it, no?

Also, `data` isn't padded. You need to write out the padding or you carry stale data (yes, you make sure it's nul-terminated, but still).

Reply:
> This doesn't append the sbat, it replaces it, no?

Yes, the content of the two is merged earlier by the caller, to de-dup the first line which is common

Reply:
I see. I misread `read_sbat`. Should really be called `merge_sbat`. Though, it's kinda ugly and wasteful to parse the stub twice now...

Reply:
It's not a performance critical code path, so it shouldn't matter too much

Reply:
> Also, data isn't padded. You need to write out the padding or you carry stale data (yes, you make sure it's nul-terminated, but still).

Added, please double check that I've not done something stoopid